components/kserve/kserve.go

```diff
@@ -166,6 +183,43 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client, resC
 	return nil
 }
 
+func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	inferenceServiceConfigMap := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{
+		Namespace: dscispec.ApplicationsNamespace,
+		Name:      "inferenceservice-config",
+	}, inferenceServiceConfigMap)
+	if err != nil {
+		fmt.Println("error to get configmap 'inferenceservice-config'")
+		return err
+	}
+
+	//set data.deploy.defaultDeploymentMode to the model specified in the Kserve spec
+	var deployData map[string]interface{}
+	json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData)
+	deployData["defaultDeploymentMode"] = k.DefaultDeploymentMode
+	deployDataBytes, _ := json.MarshalIndent(deployData, "", " ")
+	inferenceServiceConfigMap.Data["deploy"] = string(deployDataBytes)
+
+	//if the default deployment mode is RawDeployment, also set the ingressClass to "openshift-default", otherwise it should be "istio"
```

Comment:
I think the enum you are proposing can be done for the ingress class. For the defaultDeployment enum I was thinking it would be good to match the enum with the enum that Kserve directly expects. This way we can directly pass down the value specified in the DSC, and having the deployment mode enum be consistent might also be less confusing to the user

Reply:
> I think the enum you are proposing can be done for the ingress class.

Oh yeah, that makes more sense! If you could incorporate it as part of this PR that would be amazing.

> Kserve directly expects

+1000 I missed this aspect :)

Reply:
Based on some ongoing updates in the `kserve` repo. @danielezonca @israel-hdez recommended that instead of setting a value for `IngressClassName` we would rather set `disableIngressCreation` to `true`. Therefore I removed the logic to modify the `IngressClassName`. 
https://github.com/kserve/kserve/pull/3436/files